sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/share_options.hpp

```diff
@@ -662,22 +662,17 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
      * @brief Specify the crc64 calculated for the range of bytes that must be read from the copy
      * source.
      */
-    Azure::Core::Nullable<std::string> SourceContentCrc64;
+    Azure::Core::Nullable<ContentHash> SourceContentHash;
 
     /**
-     * @brief Specify the crc64 value to operate only on range with a matching crc64 checksum.
+     * @brief Specify the access condition for the source. Only Crc64 is supported.
      */
-    Azure::Core::Nullable<std::string> SourceIfMatchCrc64;
+    ContentHashAccessConditions SourceContentHashAccessCondition;
 
     /**
-     * @brief Specify the crc64 value to operate only on range without a matching crc64 checksum.
+     * @brief The operation will only succeed if the lease access condition is met.
      */
-    Azure::Core::Nullable<std::string> SourceIfNoneMatchCrc64;
-
-    /**
-     * @brief The operation will only succeed if the access condition is met.
-     */
-    LeaseAccessConditions AccessConditions;
+    Storage::LeaseAccessConditions LeaseAccessConditions;
```

Comment:
`AccessConditions`?

Naming it `LeaseAccessConditions` will cause that versioning issue. What if in the future we support more access conditions?